python/rapidsmpf/rapidsmpf/streaming/core/fanout.pyx

```diff
@@ -0,0 +1,110 @@
+        sending the current one, ensuring backpressure and synchronized flow.
+    UNBOUNDED : int
+        Forward messages without enforcing backpressure.
+        In this mode, messages may be accumulated internally before being
+        broadcast, or they may be forwarded immediately depending on the
+        implementation and downstream consumption rate.
+
+        This mode disables coordinated backpressure between outputs, allowing
+        consumers to process at independent rates, but can lead to unbounded
+        buffering and increased memory usage.
+
+        Note: Consumers might not receive any messages until *all* upstream
+        messages have been sent, depending on the implementation and buffering
+        strategy.
+    """
+    BOUNDED = <uint8_t>cpp_FanoutPolicy.BOUNDED
+    UNBOUNDED = <uint8_t>cpp_FanoutPolicy.UNBOUNDED
+
+
+def fanout(Context ctx, Channel ch_in, list chs_out, policy):
+    """
+    Broadcast messages from one input channel to multiple output channels.
+
+    The node continuously receives messages from the input channel and forwards
+    them to all output channels according to the selected fanout policy.
+
+    Each output channel receives a shallow copy of the same message; no payload
+    data is duplicated. All copies share the same underlying payload, ensuring
+    zero-copy broadcast semantics.
+
+    Parameters
+    ----------
+    ctx : Context
+        The node context to use.
+    ch_in : Channel
+        Input channel from which messages are received.
+    chs_out : list[Channel]
+        Output channels to which messages are broadcast.
+    policy : FanoutPolicy
+        The fanout strategy to use (see FanoutPolicy).
+
+    Returns
+    -------
+    CppNode
+        Streaming node representing the fanout operation.
+
+    Raises
+    ------
+    ValueError
+        If an unknown fanout policy is specified.
+
+    Notes
+    -----
+    Since messages are shallow-copied, releasing a payload (``release<T>()``)
+    is only valid on messages that hold exclusive ownership of the payload.
+
+    Examples
+    --------
+    >>> import rapidsmpf.streaming.core as streaming
+    >>> ctx = streaming.Context(...)
+    >>> ch_in = ctx.create_channel()
+    >>> ch_out1 = ctx.create_channel()
+    >>> ch_out2 = ctx.create_channel()
+    >>> node = streaming.fanout(
+    ...     ctx, ch_in, [ch_out1, ch_out2], streaming.FanoutPolicy.BOUNDED
+    ... )
+    """
+    # Validate policy
+    if not isinstance(policy, (FanoutPolicy, int)):
+        raise TypeError(f"policy must be a FanoutPolicy enum value, got {type(policy)}")
+
+    cdef vector[shared_ptr[cpp_Channel]] _chs_out
+    owner = []
+    for ch_out in chs_out:
+        if not isinstance(ch_out, Channel):
+            raise TypeError("All elements in chs_out must be Channel instances")
+        owner.append(ch_out)
+        _chs_out.push_back((<Channel>ch_out)._handle)
+
+    cdef cpp_FanoutPolicy _policy = <cpp_FanoutPolicy>(<int>policy)
```

Comment:
If you8 use the re-exported cdef enum this is unnecessary. Check some of the pylibcudf cython code to see how it's done there.